pkgs/applications/misc/ytdlp-gui/default.nix

```diff
@@ -0,0 +1,63 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, pkgs
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "ytdlp-gui";
+  version = "0.2.5";
+
+  src = fetchFromGitHub {
+    owner = "BKSalman";
+    repo = "ytdlp-gui";
+    rev = "v${version}";
+    sha256 = "sha256-BB2zM0EIPYkv5tMX6TCXVkLoz7HQCXVMOiEEwPJRAB0=";
+  };
+
+  cargoSha256 = "sha256-1jvCcHDfMbVxsNI1iBngx8uEUkENeM7sbzSL0DgpK+o=";
+
+  nativeBuildInputs = with pkgs; [
+    pkg-config
+    cmake
+    makeWrapper
+  ];
+
+  buildInputs = with pkgs; [
+    fontconfig
+
+    vulkan-headers
+    vulkan-loader
+    libGL
+
+    libxkbcommon
+    # WINIT_UNIX_BACKEND=wayland
+    wayland
+
+    # WINIT_UNIX_BACKEND=x11
+    xorg.libXcursor
+    xorg.libXrandr
+    xorg.libXi
+    xorg.libX11
+  ];
+
+  postInstall = ''
+    for _size in "16x16" "32x32" "48x48" "64x64" "128x128" "256x256"; do
+        install -Dm644 "$src/data/icons/$_size/apps/ytdlp-gui.png" "$out/share/icons/hicolor/$_size/apps/ytdlp-gui.png"
+    done
+    install -Dm644 "$src/data/applications/ytdlp-gui.desktop" -t "$out/share/applications/"
+  '';
+
+  postFixup = ''
+    wrapProgram $out/bin/ytdlp-gui \
+      --prefix PATH : ${lib.makeBinPath [ pkgs.gnome.zenity pkgs.libsForQt5.kdialog]}\
+      --suffix LD_LIBRARY_PATH : ${pkgs.lib.makeLibraryPath buildInputs}
```

Comment:
Can we use patchelf instead?

Reply:
how would the syntax be? haven't used it before :thinking:

Reply:
> Can we use patchelf instead?

how would the syntax be? haven't used it before

Reply:
Please take a look at the repo for examples

Reply:
> Please take a look at the repo for examples

I've looked at [this](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/backup/ludusavi/default.nix) (it uses "Iced" the same GUI library I'm using, but it's using a different renderer), but it doesn't work, even after adding `vulkan-loader`, and  `vulkan-headers`, probably because `Vulkan` isn't being loaded or something like that, do you have a solution for that?

also, could you please explain the benefit of using `patchelf` over `wrapProgram`?

Reply:
> also, could you please explain the benefit of using `patchelf` over `wrapProgram`?

Those two tools try to achieve different things, the problem is that LD_LIBRARY_PATH should be avoided as much as possible because it "taints" programs started from the wrapped program with the variable, too which can cause problems.